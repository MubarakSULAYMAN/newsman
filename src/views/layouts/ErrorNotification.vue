<template>
  <v-alert
    :value="isError"
    border="left"
    dismissible
    elevation="24"
    prominent
    :type="errorType"
    transition="fab-transition"
    class="error-notification"
  >
    {{ errorInfo }} <br v-if="notEmpty" />
    {{ errorMessage }}
  </v-alert>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'ErrorNotification',

  computed: {
    notEmpty() {
      return this.errorInfo !== '' && this.errorMessage !== '';
    },

    ...mapGetters(['isError', 'errorType', 'errorMessage', 'errorInfo']),
  },
};
</script>

<style scoped>
.error-notification {
  position: fixed;
  bottom: 20vh;
  right: 1vw;
  width: 50vw;
}
</style>
