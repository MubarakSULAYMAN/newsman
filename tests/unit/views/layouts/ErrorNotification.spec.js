import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import ErrorNotification from '@/views/layouts/ErrorNotification.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Error Noification', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  test('It displays "" for errorInfo and "Error message expected Here." for errorMessage.', () => {
    const $store = {
      getters: {
        errorInfo: () => '',
        errorMessage: () => 'Error message expected Here.',
      },
    };

    const wrapper = mount(ErrorNotification, {
      localVue,
      vuetify,
      mocks: {
        $store,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
