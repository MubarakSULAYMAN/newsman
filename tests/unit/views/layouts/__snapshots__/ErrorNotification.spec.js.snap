// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Noification It displays "" for errorInfo and "Error message expected Here." for errorMessage. 1`] = `
<transition-stub name="fab-transition" class="error-notification">
  <div role="alert" class="v-alert v-sheet theme--dark elevation-24 v-alert--border v-alert--prominent v-alert--border-left error">
    <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--dark"></i>
      <div class="v-alert__content">
        () =&gt; '' <br>
        () =&gt; 'Error message expected Here.'
      </div>
      <div class="v-alert__border v-alert__border--left"></div><button type="button" class="v-alert__dismissible v-btn v-btn--icon v-btn--round theme--dark v-size--small" aria-label="Close"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close-circle theme--dark"></i></span></button>
    </div>
  </div>
</transition-stub>
`;
